### ISSUE TITLE
no-invalid-this ignores @type JSDoc tag

### ISSUE THREAD

**Issue by MaceWindu** [palantir/tslint#4664]
### Bug Report

- __TSLint version__: 5.15.0
- __TypeScript version__: .3.4.3
- __Running TSLint via__: grunt-tslint

#### ~TypeScript~ JavaScript code being linted

```js
/**
 * @this {some.type}
 * */
some.type.prototype.someMethod = function () {
    this.x = 1;
}
```

with `tslint.json` configuration:

```json
{
    jsRules: {
        "no-invalid-this": true
    }
}
```

#### Actual behavior
`no-invalid-this  the "this" keyword is disallowed outside of a class body`

#### Expected behavior
no `no-invalid-this` errors

https://github.com/Microsoft/TypeScript/wiki/JSDoc-support-in-JavaScript

**Comment by JoshuaKGoldberg** [comment]
Interesting! For TypeScript files this would be a wontfix, as you could use the `function (this: some.type) {` syntax. For *JavaScript* files only, it should be allowed to use `@this`. Good find; thanks for filing @MaceWindu!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
